src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -113,103 +114,84 @@ def compilation_hash(otf_closure: stages.ProgramCall) -> int:
     )
 
 
-GTFN_DEFAULT_TRANSLATION_STEP: step_types.TranslationStep[
-    languages.NanobindSrcL, languages.LanguageWithHeaderFilesSettings
-] = gtfn_module.GTFNTranslationStep(
-    enable_itir_transforms=True,
-    use_imperative_backend=False,
-    device_type=core_defs.DeviceType.CPU,
-)
+class GTFNCompileWorkflowFactory(factory.Factory):
+    class Meta:
+        model = recipes.OTFCompileWorkflow
 
-GTFN_GPU_TRANSLATION_STEP: step_types.TranslationStep[
-    languages.NanobindSrcL, languages.LanguageWithHeaderFilesSettings
-] = gtfn_module.GTFNTranslationStep(
-    enable_itir_transforms=True,
-    use_imperative_backend=False,
-    device_type=core_defs.DeviceType.CUDA,
-)
+    class Params:
+        device_type: core_defs.DeviceType = core_defs.DeviceType.CPU
 
-GTFN_DEFAULT_COMPILE_STEP: step_types.CompilationStep = compiler.Compiler(
-    cache_strategy=cache.Strategy.SESSION, builder_factory=compiledb.CompiledbFactory()
-)
+    translation = factory.SubFactory(
+        gtfn_module.GTFNTranslationStepFactory, device_type=factory.SelfAttribute("..device_type")
+    )
+    bindings: workflow.Workflow[
+        stages.ProgramSource, stages.CompilableSource
+    ] = nanobind.bind_source
+    compilation = factory.SubFactory(
+        compiler.CompilerFactory,
+        cache_strategy=cache.Strategy.SESSION,
+        builder_factory=compiledb.CompiledbFactory(),
+    )
+    decoration = functools.partial(
+        convert_args, device=factory.LazyAttribute(lambda o: o.device_type)
+    )
 
 
-GTFN_DEFAULT_WORKFLOW = recipes.OTFCompileWorkflow(
-    translation=GTFN_DEFAULT_TRANSLATION_STEP,
-    bindings=nanobind.bind_source,
-    compilation=GTFN_DEFAULT_COMPILE_STEP,
-    decoration=convert_args,
-)
+class GTFNBackendFactory(factory.Factory):
+    class Meta:
+        model = otf_compile_executor.OTFBackend
 
+    class Params:
+        device_name = "cpu"
+        cached_name = ""
+        gpu = factory.Trait(
+            allocator=next_allocators.StandardGPUFieldBufferAllocator(),
+            device_type=core_defs.DeviceType.CUDA,
+            device_name="gpu",
+        )
+        cached = factory.Trait(
+            executor=factory.LazyAttribute(
+                lambda o: otf_compile_executor.CachedOTFCompileExecutor(
+                    otf_workflow=workflow.CachedStep(o.otf_workflow, hash_function=o.hash_function),
+                    name=o.name,
+                )
+            ),
+            cached_name="_cached",
+        )
+        device_type = core_defs.DeviceType.CPU
+        hash_function = compilation_hash
+        otf_workflow = factory.SubFactory(
+            GTFNCompileWorkflowFactory, device_type=factory.SelfAttribute("..device_type")
+        )
+        name = factory.LazyAttribute(lambda o: f"run_gtfn_{o.device_name}{o.cached_name}")
```

Comment:
Maybe we do not want to construct a name, do we? We should just use the object created by the factory.

Reply:
The factory has to give a name to the object it creates. In practice the name attribute of the executor is mainly used in pytest to see at a glance which test was parametrized with what type of gtfn backend.